mm2src/mm2_main/src/database/stats_swaps.rs

```diff
@@ -318,10 +317,79 @@ fn insert_stats_taker_swap_sql_init(swap: &TakerSavedSwap) -> Option<(&'static s
         swap_data.maker_amount.to_string(),
         swap_data.taker_amount.to_string(),
         (is_success as u32).to_string(),
-    ];
+    ]
+    .into_iter()
+    .map(SqlValue::from)
+    .collect();
+
     Some((INSERT_STATS_SWAP_ON_INIT, params))
 }
 
+/// Constructs the update query for optional fields of the swap.
+/// These are fields which are usually sent unilaterally from one side and combined together based on the shared UUID.
+fn update_optional_info(swap: &SavedSwap) -> Option<(String, OwnedSqlNamedParams)> {
+    let mut extra_args = Vec::new();
+    let mut params = OwnedSqlNamedParams::new();
+
+    if let Some(maker_pubkey) = swap.maker_pubkey() {
+        match maker_pubkey {
+            Ok(maker_pubkey) => {
+                extra_args.push("maker_pubkey = :maker_pubkey");
+                params.push((":maker_pubkey", maker_pubkey.into()));
+            },
+            Err(e) => {
+                error!("[{}] Error on getting maker_pubkey for stats: {}", swap.uuid(), e);
+            },
+        }
+    }
+    if let Some(taker_pubkey) = swap.taker_pubkey() {
+        match taker_pubkey {
+            Ok(taker_pubkey) => {
+                extra_args.push("taker_pubkey = :taker_pubkey");
+                params.push((":taker_pubkey", taker_pubkey.into()));
+            },
+            Err(e) => {
+                error!("[{}] Error on getting taker_pubkey for stats: {}", swap.uuid(), e);
+            },
+        }
+    }
```

Comment:
nit: should we check for cleared pubkeys here and elsewhere and write them as null instead?

Reply:
we can, but i wanted them as zeros to distinguish between pubkeys that are hidden vs pubkeys that aren't sent all together.